pkg/handlers/generic/mutation/controlplanevirtualip/inject.go

```diff
@@ -187,3 +167,36 @@ func (h *ControlPlaneVirtualIP) Mutate(
 		},
 	)
 }
+
+func deleteFiles(obj *controlplanev1.KubeadmControlPlaneTemplate, filePathsToDelete ...string) {
```

Comment:
nit: 
It would be nice if the function accepts `srcFiles []bootstrapv1.File` args and keep assignment to obj.Spec.Template.Spec.KubeadmConfigSpec.Files from the returned value.
```
deleteFiles(files []bootstrapv1.File, filePathsToDelete ...string) []bootstrapv1.File
```
it can be useful to use it in `KubeadmConfigTemplate`

Reply:
Good idea, done in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1062/commits/49381076065595dd41dc82dc169e480170ab9ca1